src/Slider/components/Handle/index.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+import * as React from "react";
+import styled from "styled-components";
+import convertHexToRgba from "@kiwicom/orbit-design-tokens/lib/convertHexToRgba";
+
+import { left as leftRight } from "../../../utils/rtl";
+import defaultTheme from "../../../defaultTheme";
+
+import type { Props, CalculateLeftPosition, IsFirst } from "./index";
+
+export const calculateLeftPosition: CalculateLeftPosition = (
+  valueNow,
+  valueMin,
+  valueMax,
+  isFirst,
+  isSimple,
+) => {
+  // if first, stick it to the left edge
+  if (isFirst) {
+    // if simple (one handle and without histogram)
+    if (isSimple) {
+      return +(((valueNow - valueMin) / (valueMax - valueMin)) * 100).toFixed(1);
+    }
+    return +(((valueNow - valueMin) / (valueMax - valueMin + 1)) * 100).toFixed(1);
+  }
+  // for every other handle stick on the right edge
+  return +(((valueNow - valueMin + 1) / (valueMax - valueMin + 1)) * 100).toFixed(1);
+};
+
+export const isFirst: IsFirst = (value, valueNow, index, hasHistogram) => {
+  if (Array.isArray(value)) {
+    const max = Math.max(...value);
+    const min = Math.min(...value);
+    const maxEqualsMin = max === min;
+    const minEqualsValueNow = min === valueNow;
+    if (index !== 0 && maxEqualsMin) return false;
+    return maxEqualsMin || minEqualsValueNow;
+  }
+  return !hasHistogram;
+};
+
+const StyledHandle = styled(({ left, theme, onTop, ...props }) => <div {...props} />).attrs(
+  ({ left, onTop, theme }) => {
+    return {
+      style: {
+        // TODO: use token for deducting the half size of the Handle
+        [leftRight({ theme })]: `calc(${left}% - 12px)`,
+        zIndex: onTop ? 40 : 30,
+      },
+    };
+  },
+)`
+  display: flex;
+  position: absolute;
+  right: 0;
+  bottom: 0;
+  flex: 0 0 24px;
+  align-items: center;
+  justify-content: center;
+  box-sizing: border-box;
+  width: 24px;
```

Comment:
is it normal to hardcode pixel values in the css?
shouldn't these all be some sort of tokens for units or similar?

Reply:
It's normal for new components. We create component and then tokenize it.